src/btree/bt_vrfy.c

```diff
@@ -1168,54 +826,115 @@ __verify_txn_addr_cmp(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t cell_num,
 }
 
 /*
- * __verify_timestamp_to_pretty_string --
- *     Convert a timestamp to a pretty string, utilizes existing timestamp to string function.
+ * __verify_key_hs --
+ *     Verify a key against the history store. The unpack denotes the data store's timestamp range
+ *     information and is used for verifying timestamp range overlaps.
  */
-static const char *
-__verify_timestamp_to_pretty_string(wt_timestamp_t ts, char *ts_string)
+static int
+__verify_key_hs(
+  WT_SESSION_IMPL *session, WT_ITEM *tmp1, wt_timestamp_t newer_start_ts, WT_VSTUFF *vs)
 {
-    const char *ts_bp;
+    WT_BTREE *btree;
+    WT_CURSOR *hs_cursor;
+    WT_DECL_RET;
+    wt_timestamp_t older_start_ts, older_stop_ts;
+    uint64_t hs_counter;
+    uint32_t hs_btree_id;
+    int cmp, exact;
+    char ts_string[2][WT_TS_INT_STRING_SIZE];
 
-    switch (ts) {
-    case WT_TS_MAX:
-        ts_bp = "WT_TS_MAX";
-        break;
-    case WT_TS_NONE:
-        ts_bp = "WT_TS_NONE";
-        break;
-    default:
-        ts_bp = __wt_timestamp_to_string(ts, ts_string);
-        break;
+    btree = S2BT(session);
+    hs_btree_id = btree->id;
+
+    /*
+     * Set the data store timestamp and transactions to initiate timestamp range verification. Since
+     * transaction-ids are wiped out on start, we could possibly have a start txn-id of WT_TXN_NONE,
+     * in which case we initialize our newest with the max txn-id.
+     */
+    older_stop_ts = 0;
+
+    /*
+     * Open a history store cursor positioned at the end of the data store key (the newest record)
+     * and iterate backwards until we reach a different key or btree.
+     */
+    hs_cursor = session->hs_cursor;
+    hs_cursor->set_key(hs_cursor, hs_btree_id, tmp1, WT_TS_MAX, UINT64_MAX);
```

Comment:
Does UINT64_MAX here need to be WT_TXN_MAX?

Reply:
Good catch, that was in the original code and I didn't really look at it -- I think you're correct, we should use `WT_TXN_MAX`.

Pushed in 66ccd6594.